environment/stats/src/cloud/benchflow/collectors/stats.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	"sync"
+	"github.com/fsouza/go-dockerclient"
+	"github.com/minio/minio-go"
+)
+
+type Container struct {
+	ID           string
+	statsChannel chan *docker.Stats
+	doneChannel chan bool
```

Comment:
Try if the same approach of the stopChannel works also for the doneChannel. It depends on how the library handle this channel.